pydantic/_internal/_decorators.py

```diff
@@ -203,19 +203,20 @@ class SerializationFunctions(DecoratorFunctions[Serializer]):
 _FUNCS: set[str] = set()
 
 
-def prepare_serializer_decorator(function: Callable[..., Any], allow_reuse: bool) -> classmethod[Any]:
+_SerializerType = TypeVar('_SerializerType', bound=Callable[..., Any])
+
+
+def prepare_serializer_decorator(function: _SerializerType, allow_reuse: bool) -> _SerializerType:
     """
-    Convert the function to a classmethod if it isn't already.
     Warn about validators/serializers with duplicated names since without this, they can be overwritten silently
     which generally isn't the intended behaviour, don't run in ipython (see #312) or if `allow_reuse` is True.
     """
-    f_cls = function if isinstance(function, classmethod) else classmethod(function)
     if not allow_reuse and not in_ipython():
-        ref = f'{f_cls.__func__.__module__}::{f_cls.__func__.__qualname__}'
+        ref = f'{function.__module__}::{function.__qualname__}'
```

Comment:
This caused an issue elsewhere with partial/etc.

I think here and other places in the current v2 code we should probably not be using `__qualname__` without a fallback (if I recall correctly, the python core devs said on some issue we commented on that you shouldn't assume this exists for everything callable).

This was a v1 PR where we improved support for `partial` and similar things, probably don't want to add a regression: https://github.com/pydantic/pydantic/pull/5126/files

If you don't want to fix this as part of this PR I don't think we necessarily have to do it here, BUT then we should make sure we are tracking this somewhere.

Reply:
I agree we may need to be more careful but the reason I am removing this in this PR is that previously this function was getting wrapped with `classmethod` right above, and `classmethod` exposes the original function via `__func__`. So this change makes us no better or worse at handling `partial` than we were before.